### PR TITLE
Clean up .env File

### PR DESCRIPTION
**Pull request by MSevey**
# PULL REQUEST

## Overview
Clean up .env file, move all defaults to `playbooks/group_vars/webportals.yml` and only have hardcoded values and blanks in the `.env` file. Update comments to reflect changes.

## Checklist
Review and complete the checklist to ensure that the PR is complete before assigned to an approver.
 - [ ] All new methods or updated methods have clear docstrings
 - [ ] Testing added or updated for new methods
 - [ ] Verify if any changes impact the WebPortal Health Checks
 - [ ] Approriate documentation updated
 - [ ] Changelog file created

### CODE REVIEW

File: playbooks/templates/.env.j2 [hunk 1]
```diff
@@ -47,42 +55,56 @@ PORTAL_NAME={{ webportal_server_config.portal_name }}
...
 # Sia wallet password
+#
+# REQUIRED: Ansible automatically generated 
 SIA_WALLET_PASSWORD={{ webportal_server_config.sia_wallet_password if (webportal_server_config.sia_wallet_password is defined and webportal_server_config.sia_wallet_password is not none and webportal_server_config.sia_wallet_password != '') else (webportal_server_config.sia_wallet_seed if webportal_server_config.sia_wallet_seed is defined else '') }}
 
-# (optional) The port on which siad is listening, defaults to 9980.
+# The port on which siad is listening, defaults to 9980. 
+#
+# OPTIONAL: manually defined by user, or default will be used
+# TODO: Does this default need to be explicitly set?
```
Comment (ro-tex) [inline playbooks/templates/.env.j2:75]: API_PORT is only used by `blocker`, if I'm not mistaken, and it doesn't need it to be defined - it will happily use 9980 otherwise.

File: playbooks/templates/.env.j2 [hunk 2]
```diff
@@ -107,64 +133,96 @@ AIRTABLE_FIELD={{ webportal_common_config.airtable_field }}
...
 # MongoDB password
+#
+# REQUIRED: Ansible automatically generated 
+# TODO: Ansible doesn't not currently automatically generate this but it should 
 SKYNET_DB_PASS={{ mongodb_config.skynet_db_pass }}
 
 # MongoDB address or container name
+#
+# OPTIONAL: manually defined by user, or default will be used
+# TODO: Do these need to be able to be defined by the user?
```
Comment (ro-tex) [inline playbooks/templates/.env.j2:149]: I don't think we *need* to make this configurable. Fewer configuration options might actually be better at this stage.

File: playbooks/templates/.env.j2 [hunk 3]
```diff
@@ -107,64 +133,96 @@ AIRTABLE_FIELD={{ webportal_common_config.airtable_field }}
...
 SKYNET_DB_HOST=mongo
 
 # MongoDB port
+#
+# OPTIONAL: manually defined by user, or default will be used
+# TODO: Do these need to be able to be defined by the user?
```
Comment (ro-tex) [inline playbooks/templates/.env.j2:155]: Same as SKYNET_DB_HOST - no need.

File: playbooks/templates/.env.j2 [hunk 4]
```diff
@@ -107,64 +133,96 @@ AIRTABLE_FIELD={{ webportal_common_config.airtable_field }}
...
 
 # Absolute url to the (cluster) portal dashboard.
+#
+# REQUIRED: See top of file 
 SKYNET_DASHBOARD_URL=https://account.{{ cluster_domain }}
 
 # Accounts log level
+#
+# OPTIONAL: manually defined by user, or default will be used
+# TODO: Do these need to be able to be defined by the user?
```
Comment (ro-tex) [inline playbooks/templates/.env.j2:175]: All log levels should be configurable by the user. They are very helpful when debugging an issue. If a portal op reaches out to us on Discord and asks us about an issue, one of the first things I would do is ask them for their logs with `level=trace`, so we can see detailed logging.

File: playbooks/templates/.env.j2 (hunk 1 above)
Comment (MSevey) [reply playbooks/templates/.env.j2:75]: we should just remove this then.

File: playbooks/templates/.env.j2 (hunk 3 above)
Comment (MSevey) [reply playbooks/templates/.env.j2:155]: same question

**Comment by MSevey** [comment]
Made some updates, added some more defaults so most things can be configurable, but don't need to be

**Review by kwypchlo — DISMISSED** [review]

**Review by firyx — APPROVED** [review]

File: playbooks/group_vars/webportals.yml [hunk 5]
```diff
@@ -113,16 +113,24 @@ webportal_allowance:
 webportal_server_config_defaults:
   domain_name: "{{ ansible_host }}"
   portal_name: "{{ ansible_host }}"
-  portal_modules: ""
+  portal_modules: "m"
```
Comment (kwypchlo) [inline playbooks/group_vars/webportals.yml:116]: do we want to also include accounts by default or we want to default to as minimal setup as possible? including just mongo sounds redundant

File: playbooks/group_vars/webportals.yml [hunk 6]
```diff
@@ -113,16 +113,24 @@ webportal_allowance:
...
   sia_wallet_password: ""
   sia_api_port: 9980
+  sia_api_password: "{{ lookup('pipe','openssl rand -base64 32') }}"
```
Comment (kwypchlo) [inline playbooks/group_vars/webportals.yml:119]: is this gonna work? sia_api_password should equal the one that skyd generates on the first run right?

File: playbooks/templates/.env.j2 [hunk 7]
```diff
@@ -138,63 +179,85 @@ SKYNET_ACCOUNTS_LOG_LEVEL=info
...
 ACCOUNTS_LIMIT_ACCESS={{ webportal_common_config.accounts_limit_access }}
 
 {% endif %}
 # Accounts email
+# REQUIRED: manually defined by user
 ACCOUNTS_EMAIL_URI={{ webportal_common_config.accounts_email_uri }}
 
 # Domain to which your cookies will be issued
+#
+# REQUIRED: See top of file
```
Comment (kwypchlo) [inline playbooks/templates/.env.j2:192]: do we want to note that vars like this are required only if accounts module (or some other module) is enabled?

File: playbooks/group_vars/webportals.yml (hunk 6 above)
Comment (MSevey) [reply playbooks/group_vars/webportals.yml:119]: yup, skyd checks for an environment variable first. So even if it previously created one on disk, when it restarts it will start using the new env var.

File: playbooks/group_vars/webportals.yml (hunk 5 above)
Comment (MSevey) [reply playbooks/group_vars/webportals.yml:116]: I went with the minimum, so no accounts. Since our docker compose files set the mongo env vars for skyd, skyd won't start without mongo.  So even though mongo isn't required for a bare bones skyd instance, because ansible is using our docker compose files, mongo becomes required. 

tl;dr minimum set up requires mongo.

File: playbooks/templates/.env.j2 (hunk 7 above)
Comment (MSevey) [reply playbooks/templates/.env.j2:192]: this is already grouped like that :-)

This is under the `(optional) Accounts` section.

**Review by firyx — APPROVED** [review]

File: playbooks/group_vars/webportals.yml (hunk 5 above)
Comment (kwypchlo) [reply playbooks/group_vars/webportals.yml:116]: > Since our docker compose files set the mongo env vars for skyd, skyd won't start without mongo

This is not correct any more, I fixed that couple of days ago when I tried starting without mongo and it worked fine :) https://github.com/SkynetLabs/skynet-webportal/commit/dba33f833569968b8656ca8b53a3cec5d7090caf

File: playbooks/group_vars/webportals.yml (hunk 6 above)
Comment (kwypchlo) [reply playbooks/group_vars/webportals.yml:119]: This will break nginx reading api password since it expects it to be stored on a disk in our setup.

https://github.com/SkynetLabs/skynet-webportal/blob/master/docker/nginx/libs/skynet/utils.lua

File: playbooks/templates/.env.j2 [hunk 8]
```diff
@@ -95,41 +118,59 @@ DISCORD_MENTION_USER_ID={{ discord_mention_user_id | default('') }}
...
 # MongoDB password
+#
+# REQUIRED: manually defined by user, or Ansible automatically generated 
 SKYNET_DB_PASS={{ mongodb_config.skynet_db_pass }}
 
 # MongoDB address or container name
-SKYNET_DB_HOST=mongo
+#
+# REQUIRED: manually defined by user, or default will be used
+SKYNET_DB_HOST={{ mongodb_config.skynet_db_host }}
```
Comment (firyx) [inline playbooks/templates/.env.j2:149]: Is this configurable for all services using mongo?
Isn't `mongo` container service/host name hardcoded in some docker services using mongo?

**Review by firyx — CHANGES_REQUESTED** [review]

File: playbooks/group_vars/webportals.yml (hunk 6 above)
Comment (MSevey) [reply playbooks/group_vars/webportals.yml:119]: Plan is to update nginx

File: playbooks/templates/.env.j2 (hunk 8 above)
Comment (MSevey) [reply playbooks/templates/.env.j2:149]: There are two hardcoded references 

https://github.com/SkynetLabs/skynet-webportal/blob/master/docker/malware-scanner/Dockerfile#L16
https://github.com/SkynetLabs/skynet-webportal/blob/master/docker/accounts/Dockerfile#L16

But then they are overwritten in the docker-compose files

https://github.com/SkynetLabs/skynet-webportal/blob/master/docker-compose.accounts.yml#L40

So since nothing is hardcoded in the docker compose files, this means that they are configurable right?

File: playbooks/group_vars/webportals.yml (hunk 5 above)
Comment (MSevey) [reply playbooks/group_vars/webportals.yml:116]: Nice 👍 

I think we can leave this for now. If we want to change it in the future we can.

File: playbooks/group_vars/webportals.yml (hunk 6 above)
Comment (kwypchlo) [reply playbooks/group_vars/webportals.yml:119]: resolved in https://github.com/SkynetLabs/skynet-webportal/pull/1567

File: playbooks/group_vars/webportals.yml (hunk 5 above)
Comment (MSevey) [inline playbooks/group_vars/webportals.yml:116]:
```suggestion
  portal_modules: ""
```

**Review by firyx — APPROVED** [review]

**Review by kwypchlo — APPROVED** [review]
